src/daemon/rpmostree-sysroot-upgrader.cxx

```diff
@@ -450,28 +450,19 @@ rpmostree_sysroot_upgrader_pull_base (RpmOstreeSysrootUpgrader  *self,
         if (override_commit)
           return glnx_throw (error, "Specifying commit overrides for container-image-reference type refspecs is not supported");
 
-        if (strstr (refspec, "@"))
-          return glnx_throw (error, "Pinned to specific container image digest, cannot upgrade");
-
-        /* Check to see if the digest that `refspec` points to is the same before pulling a new container image. */
-        const char *cur_digest = rpmostree_origin_get_container_image_reference_digest (self->original_origin);
-        if (cur_digest)
+        auto refspec_s = std::string(refspec);
+        auto import = CXX_TRY_VAL(pull_container(*self->repo, *cancellable, refspec_s), error);
+        if (!import->changed) 
```

Comment:
Does this change detection logic correctly handle the case where a newer image was already pulled but not yet deployed?

Reply:
Huh, good catch.  I think the bug here dates to 7ac7ef4d32ff92814f54f01a281aacc9d5dee90d

AFAICS, the `return` there was always unnecessary.  But testing now.

Reply:
OK, actually being able to remove that in turn means we can remove the now-unnecessary `changed` boolean which in turn means we can drop that wrapper struct, etc.  :tada: